### PR TITLE
Use grid area code instead of id

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

The grid area code is unique as well as the id. Currently, we don't know the ids in the UI, so for now it's simpler to stick with the codes.

## References

* #33

### CODE REVIEW

File: source/dotnet/Application/Batches/BatchRequestDto.cs [hunk 1]
```diff
@@ -13,11 +13,15 @@
...
-using Energinet.DataHub.Wholesale.Domain.GridAreaAggregate;
 
-namespace Energinet.DataHub.Wholesale.Application;
+namespace Energinet.DataHub.Wholesale.Application.Batches;
 
-public interface IBatchApplicationService
+/// <summary>
+/// A request to create a batch.
+/// </summary>
+public class BatchRequestDto
```
Comment (defectiveAi) [inline source/dotnet/Application/Batches/BatchRequestDto.cs:22]: Data from a request are immutable, so this dto could be too.

File: source/dotnet/Domain/GridAreaAggregate/GridAreaCode.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+using System.Text.RegularExpressions;
+
+namespace Energinet.DataHub.Wholesale.Domain.GridAreaAggregate;
+
+public sealed record GridAreaCode
+{
+    public GridAreaCode(string code)
+    {
+        if (code == null) throw new ArgumentNullException(nameof(code));
```
Comment (defectiveAi) [inline source/dotnet/Domain/GridAreaAggregate/GridAreaCode.cs:23]: ArgumentNullException.ThrowIfNull(code)

File: source/dotnet/Domain/GridAreaAggregate/GridAreaCode.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        if (!Regex.IsMatch(code, @"^((00\d)|(0[1-9]\d)|([1-9]\d\d))$", RegexOptions.ECMAScript))
```
Comment (defectiveAi) [inline source/dotnet/Domain/GridAreaAggregate/GridAreaCode.cs:24]: Very nice!

File: source/dotnet/Domain/GridAreaAggregate/GridAreaCode.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/dotnet/Domain/GridAreaAggregate/GridAreaCode.cs:23]: Nice. I didn't know that method.

File: source/dotnet/Application/Batches/BatchRequestDto.cs [hunk 4]
```diff
@@ -13,11 +13,10 @@
...
+namespace Energinet.DataHub.Wholesale.Application.Batches;
 
-public interface IBatchApplicationService
-{
-    Task CreateAsync(WholesaleProcessType processType, IEnumerable<GridAreaId> gridAreas);
-}
+/// <summary>
+/// An immutable request to create a batch.
+/// </summary>
+public sealed record BatchRequestDto(WholesaleProcessType ProcessType, List<string> GridAreaCodes);
```
Comment (defectiveAi) [inline source/dotnet/Application/Batches/BatchRequestDto.cs:22]: Technically, list is mutable.

Comment (BjarkeMeier) [reply source/dotnet/Application/Batches/BatchRequestDto.cs]: You're right. Changed to IEnumerable.

**Pull request merged**